fabric-gametest-api-v1/src/main/java/net/fabricmc/fabric/mixin/gametest/StructureTemplateManagerMixin.java

```diff
@@ -0,0 +1,90 @@
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+import net.minecraft.block.Block;
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.nbt.NbtHelper;
+import net.minecraft.registry.RegistryEntryLookup;
+import net.minecraft.resource.Resource;
+import net.minecraft.resource.ResourceFinder;
+import net.minecraft.resource.ResourceManager;
+import net.minecraft.structure.StructureTemplate;
+import net.minecraft.structure.StructureTemplateManager;
+import net.minecraft.util.Identifier;
+import net.minecraft.world.level.storage.LevelStorage;
+
+import net.fabricmc.fabric.impl.gametest.FabricGameTestHelper;
+
+@Mixin(StructureTemplateManager.class)
+public abstract class StructureTemplateManagerMixin {
+	@Shadow
+	private ResourceManager resourceManager;
+
+	@Shadow
+	public abstract StructureTemplate createTemplate(NbtCompound nbt);
+
+	@Unique
+	private final Logger MIXIN_LOGGER = LoggerFactory.getLogger("StructureTemplateManagerMixin");
+
+	private Optional<StructureTemplate> fabric_loadSnbtFromResource(Identifier id) {
+		Identifier path = FabricGameTestHelper.GAMETEST_STRUCTURE_FINDER.toResourcePath(id);
+		Optional<Resource> resource = this.resourceManager.getResource(path);
+
+		if (resource.isPresent()) {
+			try {
+				String snbt = IOUtils.toString(resource.get().getReader());
+				NbtCompound nbt = NbtHelper.fromNbtProviderString(snbt);
+				return Optional.of(this.createTemplate(nbt));
+			} catch (IOException | CommandSyntaxException e) {
+				MIXIN_LOGGER.error("Failed to load GameTest structure {}", id, e);
+			}
+		}
+
+		return Optional.empty();
+	}
+
+	private Stream<Identifier> fabric_streamTemplatesFromResource() {
+		ResourceFinder finder = FabricGameTestHelper.GAMETEST_STRUCTURE_FINDER;
+		return finder.findResources(this.resourceManager).keySet().stream().map(finder::toResourceId);
+	}
+
+	@Inject(method = "<init>", at = @At(value = "INVOKE", target = "Lcom/google/common/collect/ImmutableList$Builder;add(Ljava/lang/Object;)Lcom/google/common/collect/ImmutableList$Builder;", ordinal = 2, shift = At.Shift.AFTER), locals = LocalCapture.CAPTURE_FAILHARD)
+	private void addFabricTemplateProvider(ResourceManager resourceManager, LevelStorage.Session session, DataFixer dataFixer, RegistryEntryLookup<Block> blockLookup, CallbackInfo ci, ImmutableList.Builder<StructureTemplateManager.Provider> builder) {
+		if (FabricGameTestHelper.ENABLED) {
+			builder.add(new StructureTemplateManager.Provider(this::fabric_loadSnbtFromResource, this::fabric_streamTemplatesFromResource));
+		}
```

Comment:
I believe this should always add the new provider as this will break when running the tests via `/test runall` from a normal client launch.

`FabricGameTestHelper.ENABLED` is a marker for when the normal server startup should be hyjacked to run the test server.